### PR TITLE
Task 4: Add GET /api/articles/:article_id endpoint

### PR DESCRIPTION
**Pull request by NicholasCanny**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -48,3 +48,59 @@ describe("GET /api/topics", () => {
...
+          title: "Living in the shadow of a great man",
+          topic: "mitch",
+          author: "butter_bridge",
+          body: "I find this existence challenging",
+          created_at: expect.any(String), // As this is time stamped, instead checks that created_at is a string
+          votes: 100,
+          article_img_url:
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+        });
+      });
```
Comment (nc-mark-t) [inline __tests__/app.test.js:84]: Should this have an article_id?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -48,3 +48,59 @@ describe("GET /api/topics", () => {
...
+  });
+  test("should respond with bad request if invalid type inputted for Id", () => {
+    return request(app)
+      .get("/api/articles/banana")
+      .expect(400)
+      .then((response) => {
+        expect(response.body).toEqual({
+          error: "Bad Request",
+        });
```
Comment (nc-mark-t) [inline __tests__/app.test.js:103]: Nice edge cases and error handling

File: app.js [hunk 3]
```diff
@@ -2,22 +2,41 @@ const express = require("express");
...
+app.use((err, req, res, next) => {
+  if (err.code === "22P02" || err.code === "23502") {
+    res.status(400).send({ error: "Bad Request" });
+  } else {
+    next(err);
+  }
+});
+
+app.use((err, req, res, next) => {
+  if (err.message === "article not found") {
```
Comment (nc-mark-t) [inline app.js:28]: As the app grows you may just want to check for err.message and respond with that, making it more dynamic

File: model.js [hunk 4]
```diff
@@ -6,4 +6,18 @@ const fetchTopics = () => {
   });
 };
 
-module.exports = fetchTopics;
+const fetchArticleByArticleID = (id) => {
+  return db
+    .query(`SELECT * FROM articles WHERE article_id=$1`, [id]) // subsitution - $1 == first element in the array
+    .then(({ rows }) => {
+      if (rows.length === 0) {
```
Comment (nc-mark-t) [inline model.js:13]: You could use `rowCount` here

Comment (nc-mark-t) [reply model.js:13]: I don't think the comments add any additional context so I would remove them

**Review by nc-mark-t — COMMENTED** [review]
👍
